include/zephyr/net/net_pkt.h

```diff
@@ -123,6 +123,17 @@ struct net_pkt {
 	struct net_if *orig_iface; /* Original network interface */
 #endif
 
+#if defined(CONFIG_WIREGUARD)
+	/* Original network interface */
+	struct net_if *wg_iface;
+	/* Pointer to IP header of the encrypted pkt */
+	union net_ip_header wg_ip_hdr;
+	/* Pointer to UDP header of the encrypted pkt */
+	union net_proto_header wg_proto_hdr;
+	/* Peer id */
+	int wg_peer_id;
```

Comment:
Instead of prefixing all with `wg_`, put it in a struct?

```c
	struct {
		/* Original network interface */
		struct net_if *iface;
		/* Pointer to IP header of the encrypted pkt */
		union net_ip_header ip_hdr;
		/* Pointer to UDP header of the encrypted pkt */
		union net_proto_header proto_hdr;
		/* Peer id */
		int peer_id;
	} wg;
```

Reply:
Sure, we can certainly do that.